WordPress/Classes/ViewRelated/Blog/SharingAuthorizationWebViewController.h

```diff
@@ -0,0 +1,39 @@
+#import <UIKit/UIKit.h>
+
+#import "WPWebViewController.h"
+
+@class Publicizer;
+@class Blog;
+
+/**
+ *	@brief	Results of attempted authorization
+ */
+@protocol SharingAuthorizationDelegate <NSObject>
+@optional
+- (void)authorizeDidSucceed:(Publicizer *)publicizer;
+- (void)authorize:(Publicizer *)publicizer didFailWithError:(NSError *)error;
+- (void)authorizeDidCancel:(Publicizer *)publicizer;
+@end
+
+/**
+ *	@brief	Modal controller for hosting 3rd party service (Publicize) login
+ */
+@interface SharingAuthorizationWebViewController : WPWebViewController
```

Comment:
Although the superclass wasn't originally intended for subclassing, i really like your approach. Super clean, and only relying on public methods!